### PR TITLE
[WIP] Add lazy primitives to DAML

### PR DESCRIPTION
**Pull request by hurryabit**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/6357)

### CODE REVIEW

File: compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml [hunk 1]
```diff
@@ -244,3 +250,13 @@ data TypeRep = TypeRep Opaque
 instance Eq TypeRep where
   (==) = primitive @"BEEqual"
 #endif
+
+#ifdef DAML_UNSTABLE
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml:254]: Please use a separate feature flag (e.g. `DAML_LAZY`) to gate this feature.

`DAML_UNSTABLE` is only for the unstable text functions, not a general flag for unstable features.

Comment (hurryabit) [reply compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml:254]: Yep, I'll move it under a separate flag if we decide to merge it.

Comment (cocreature) [reply compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml:254]: Maybe worth renaming it to `DAML_UNSTABLE_TEXT`?

**Comment by cocreature** [comment]
not happening for now.

**Pull request closed without merging**